### DISCUSSION TITLE
Quaternion order?

### DISCUSSION THREAD

**Discussion by namheegordonkim** [google/brax#251] [General]
https://github.com/google/brax/blob/b3e75f9f0a66c19a3d76f232218b5029f116f3dd/js/system.js#L317

Mostly FYI for those implementing their own renderers.

It seems that the quaternion coordinates used by Brax is in a different order than convention. Is there a good reason behind this?

**Comment by btaba** [comment]
Hi @namheegordonkim, yes the quat order in brax is w, x, y, z, so we do a little switcharoo to plug into THREE.js

**Reply by JoeyTeng** [reply]
With the release of [JAX 0.4.12](https://jax.readthedocs.io/en/latest/changelog.html#jax-0-4-12-june-8-2023), their quaternion order is in x, y, z, w (deduced from [code here](https://github.com/google/jax/blob/1a7336d44938e95f1baa53ace51e797df57a994a/jax/_src/scipy/spatial/transform.py#L225)) is there a plan to adapt their change and make these behaviours consistent?

**Reply by btaba** [reply]
Hi @JoeyTeng , sorry for the late reply. It looks like jax.scipy is following scipy convention, but we are unlikely to switch to x,y,z,w. We are aligning brax more with MuJoCo these days
